### PR TITLE
Restructuring of Kubernetes components section.

### PR DESCRIPTION
**Pull request by KoryKessel-Mirantis**

### CODE REVIEW

File: content/docs/operations/kubernetes/kube-apiserver.md [hunk 1]
```diff
@@ -0,0 +1,31 @@
+---
+title: kube-apiserver
+weight: 3
+---
+
+The Kubernetes API server validates and configures data for the API objects,
+which include pods, services, and replicationcontrollers, among others. The
```
Comment (MagdaDziadosz) [inline content/docs/operations/kubernetes/kube-apiserver.md:7]:
```suggestion
which include pods, services, and replication controllers, among others. The
```

File: content/docs/operations/kubernetes/kube-apiserver.md [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+server performs REST operations while also serving as the frontend to the
+shared state of a cluster, through which the other components interact.
+
+You can configure the Kubernetes API server for all controllers through the
+`apiServer` section of the MKE configuration file, an example of which follows:
```
Comment (MagdaDziadosz) [inline content/docs/operations/kubernetes/kube-apiserver.md:12]:
```suggestion
`apiServer` section of the MKE configuration file:
```

File: content/docs/operations/kubernetes/kube-controller-manager.md [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+The Kubernetes controller manager is a daemon that embeds the core control
+loops that ship with Kubernetes. In Kubernetes, a controller is a control loop
+that watches the shared state of the cluster through the kube-apiserver, making
+changes designed to move the current state towards the desired state. Examples
+of controllers that ship with Kubernetes are the replication controller,
+endpoints controller, namespace controller, and the service accounts
+controller.
+
+You can configure all Kubernetes controllers through the `controllerManager`
+section of the MKE configuration file, an example of which follows:
```
Comment (MagdaDziadosz) [inline content/docs/operations/kubernetes/kube-controller-manager.md:15]:
```suggestion
section of the MKE configuration file:
```

File: content/docs/operations/kubernetes/kube-scheduler.md [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+
+You can use multiple different schedulers within a cluster; kube-scheduler is
+the reference implementation.
+
+You can configure all Kubernetes controllers through the `scheduler` section of
+the MKE configuration file, an example of which follows:
+
+```yaml
+spec:
+  scheduler:
```
Comment (MagdaDziadosz) [inline content/docs/operations/kubernetes/kube-scheduler.md:20]: Add missing code example.

File: content/docs/operations/kubernetes/kube-apiserver.md [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+      maxAge: 30
+      maxBackup: 10
+      maxSize: 10
+    encryptionProvider: /var/lib/k0s/encryption.cfg
+    eventRateLimit:
+      enabled: false
+    requestTimeout: 1m0s
+```
+
+You can further configure kube-apiserver using the `extraArgs` field to define
```
Comment (MagdaDziadosz) [inline content/docs/operations/kubernetes/kube-apiserver.md:29]:
```suggestion
You can further configure Kubernetes API server using the `extraArgs` field to define
```

File: content/docs/operations/kubernetes/kube-controller-manager.md [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+
+```yaml
+spec:
+  controllerManager:
+    terminatedPodGCThreshold: 12500
+```
+
+You can further configure kube-controller-manager using the `extraArgs` field
```
Comment (MagdaDziadosz) [inline content/docs/operations/kubernetes/kube-controller-manager.md:23]:
```suggestion
You can further configure Kubernetes controller manager using the `extraArgs` field
```

File: content/docs/operations/kubernetes/kube-scheduler.md [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+```
+
+You can further configure kube-scheduler using the `extraArgs` field to define
```
Comment (MagdaDziadosz) [inline content/docs/operations/kubernetes/kube-scheduler.md:23]:
```suggestion
You can further configure Kubernetes scheduler using the `extraArgs` field to define
```

File: content/docs/operations/kubernetes/kubelet.md [hunk 8]
```diff
@@ -0,0 +1,120 @@
...
+
+You can also configure a kubelet with custom profiles. Such profiles offer
+greater control of the `KubeletConfiguration` and can be targeted to specific
+hosts.
+
+## kubelet custom profiles
+
+You can deploy custom profiles to configure kubelet on a per-node basis.
+
+A kubelet custom profile is comprised of a profile name and a set of values.
```
Comment (MagdaDziadosz) [inline content/docs/operations/kubernetes/kubelet.md:53]:
```suggestion
A kubelet custom profile consists of a profile name and a set of values.
```

File: content/docs/operations/kubernetes/kubelet.md [hunk 9]
```diff
@@ -0,0 +1,120 @@
...
+The profile name is used to identify the profile and to target it to specific
+nodes in the cluster, while the values are merged into the final kubelet
+configuration that is applied to a target node.
```
Comment (MagdaDziadosz) [inline content/docs/operations/kubernetes/kubelet.md:56]:
```suggestion
configuration applied to a target node.
```

File: content/docs/operations/kubernetes/kubelet.md [hunk 10]
```diff
@@ -0,0 +1,120 @@
...
+            imagefs.available: 15%
+            memory.available: 100Mi
+            nodefs.available: 10%
+            nodefs.inodesFree: 5%
+```
+
+### Apply a custom profile to a node
+
+Hosts can be assigned a custom profile through the `hosts` section of the MKE
+configuration file, whereas the profile name is an install time argument for
```
Comment (MagdaDziadosz) [inline content/docs/operations/kubernetes/kubelet.md:89]:
```suggestion
configuration file, whereas the profile name is an installation time argument for
```

File: content/docs/operations/kubernetes/kubelet.md [hunk 11]
```diff
@@ -0,0 +1,120 @@
...
+nodes in the cluster, while the values are merged into the final kubelet
+configuration that is applied to a target node.
+
+### Create a custom profile
+
+You can specify custom profiles in the `kubelet.customProfiles` section of the
+MKE configuration file. Profiles must each have a unique name, and values can
+refer to fields in the kubelet configuration file.
+
+For detail on all possible values, refer to the official Kubrernetes
```
Comment (MagdaDziadosz) [inline content/docs/operations/kubernetes/kubelet.md:64]:
```suggestion
For detail on all possible values, refer to the official Kubernetes
```

File: content/docs/operations/kubernetes/kubelet.md [hunk 12]
```diff
@@ -0,0 +1,120 @@
+---
+title: kubelet
+weight: 2
+---
+
+The kubelet component runs on each node in a Kubernetes cluster, which serves
+as the primary administrative agent for each node, monitoring application
+servers and routing administrative requests to servers. You can configure
+kubelet for all cluster nodes through the `kubelet` section of the MKE
+configuration file, an example of which follows:
```
Comment (MagdaDziadosz) [inline content/docs/operations/kubernetes/kubelet.md:10]:
```suggestion
configuration file:
```

File: content/docs/operations/kubernetes/kubelet.md [hunk 13]
```diff
@@ -0,0 +1,120 @@
...
+configuration that is applied to a target node.
+
+### Create a custom profile
+
+You can specify custom profiles in the `kubelet.customProfiles` section of the
+MKE configuration file. Profiles must each have a unique name, and values can
+refer to fields in the kubelet configuration file.
+
+For detail on all possible values, refer to the official Kubrernetes
+documentation [Set Kubelet Parameters Via a Configuration
```
Comment (MagdaDziadosz) [inline content/docs/operations/kubernetes/kubelet.md:65]:
```suggestion
documentation [Set Kubelet Parameters Via A Configuration
```

File: content/docs/operations/kubernetes/kubelet.md [hunk 14]
```diff
@@ -0,0 +1,120 @@
...
+File](https://kubernetes.io/docs/tasks/administer-cluster/kubelet-config-file/).
+
+The following example configuration creates a custom profile named `hardworker`
+that specifies thresholds for the garbage collection of images and eviction.
```
Comment (MagdaDziadosz) [inline content/docs/operations/kubernetes/kubelet.md:69]:
```suggestion
that specifies thresholds for the garbage collection of images and eviction:
```

File: content/docs/operations/kubernetes/kubelet.md [hunk 15]
```diff
@@ -0,0 +1,120 @@
...
+```
+
+### Apply a custom profile to a node
+
+Hosts can be assigned a custom profile through the `hosts` section of the MKE
+configuration file, whereas the profile name is an install time argument for
+the host.
+
+The following example configuration applies the `hardworker` custom profile to
+the `localhost` node.
```
Comment (MagdaDziadosz) [inline content/docs/operations/kubernetes/kubelet.md:93]:
```suggestion
the `localhost` node:
```

**Review by MagdaDziadosz — CHANGES_REQUESTED** [review]

File: content/docs/operations/kubernetes/kube-scheduler.md (hunk 4 above)
Comment (KoryKessel-Mirantis) [reply content/docs/operations/kubernetes/kube-scheduler.md:20]: @nsteph, is there more you can offer here for the code example?

Comment (nsteph) [inline content/docs/operations/kubernetes/kube-scheduler.md:20]:
```suggestion
  scheduler:
    profilingEnabled: true
    bindToAll: true
```

File: content/docs/operations/kubernetes/kubelet.md [hunk 16]
```diff
@@ -0,0 +1,120 @@
...
+hosts.
+
+## kubelet custom profiles
+
+You can deploy custom profiles to configure kubelet on a per-node basis.
+
+A kubelet custom profile comprises a profile name and a set of values.
+The profile name is used to identify the profile and to target it to specific
+nodes in the cluster, while the values are merged into the final kubelet
+configuration that is applied to a target node.
```
Comment (KoryKessel-Mirantis) [inline content/docs/operations/kubernetes/kubelet.md:56]:
```suggestion
A kubelet custom profile comprises a profile name and a set of values.
The profile name identifies the profile and targets it to specific
nodes in the cluster, while the values are merged into the final kubelet
configuration that is applied to a target node.
```

**Review by MagdaDziadosz — APPROVED** [review]

**Pull request merged**